moveit_ros/move_group/src/default_capabilities/move_action_capability.cpp

```diff
@@ -64,20 +65,26 @@ void move_group::MoveGroupMoveAction::executeMoveCallback(const moveit_msgs::Mov
   context_->planning_scene_monitor_->waitForCurrentRobotState(ros::Time::now());
   context_->planning_scene_monitor_->updateFrameTransforms();
 
+  // Update the goal's start state to the robot's current state using the latest data from the planning scene
+  planning_scene_monitor::LockedPlanningSceneRO lscene(context_->planning_scene_monitor_);
   moveit_msgs::MoveGroupResult action_res;
-  if (goal->planning_options.plan_only || !context_->allow_trajectory_execution_)
+  const robot_state::RobotState& current_state = lscene->getCurrentState();
+  moveit_msgs::MoveGroupGoal* new_goal = new moveit_msgs::MoveGroupGoal(*goal);
+  robot_state::robotStateToRobotStateMsg(current_state, new_goal->request.start_state);
+  moveit_msgs::MoveGroupGoalConstPtr change_goal(new_goal);
+  if (change_goal->planning_options.plan_only || !context_->allow_trajectory_execution_)
   {
-    if (!goal->planning_options.plan_only)
+    if (!change_goal->planning_options.plan_only)
       ROS_WARN("This instance of MoveGroup is not allowed to execute trajectories but the goal request has plan_only "
                "set to false. Only a motion plan will be computed anyway.");
-    executeMoveCallback_PlanOnly(goal, action_res);
+    executeMoveCallback_PlanOnly(change_goal, action_res);
   }
   else
-    executeMoveCallback_PlanAndExecute(goal, action_res);
+    executeMoveCallback_PlanAndExecute(change_goal, action_res);
```

Comment:
Please have a look inside this function.
PlanAndExecute requires the start state to be the current state, by [clearing it if it is not empty](https://github.com/ros-planning/moveit/pull/1165/files#diff-8ce3d58d373bb855098ec31aa4c588d4R129).

This contradicts the logic in this pull-request in practice.
I would propose to move this check from inside the function to the conditional I proposed above.
If the plan should be executed right away, a user-supplied start state should be ignored and a warning should be issued, as before.

Reply:
As I understand, the conditional you suggested implements the use-case when there is a user-supplied start state. Then `motion_plan_request` should be assigned directly to the value of `goal->request` in `executeMoveCallback_PlanAndExecute()` function without cleanup.

Reply:
> Why do you think its related to SRDFDom?

Only errors I got was related to it and also the build command I was using lacked it and Travis was not producing any meaningful outputs, therefore I thought it was the reason. Nonetheless, Travis can finish now so that I finally found out the problem. It is related with a test in CHOMP Planner. Below is the output of the error in Travis. The case of `noStartState` fails since it can't fetch the error value it waits for.

```ruby
<?xml version="1.0" encoding="UTF-8"?>
<testsuites tests="5" failures="1" disabled="0" errors="0" timestamp="2018-11-27T10:13:11" time="8.995" name="AllTests">
  <testsuite name="CHOMPMoveitTest" tests="5" failures="1" disabled="0" errors="0" time="8.995">
    <testcase name="jointSpaceGoodGoal" status="run" time="2.504" classname="CHOMPMoveitTest" />
    <testcase name="jointSpaceBadGoal" status="run" time="1.284" classname="CHOMPMoveitTest" />
    <testcase name="cartesianGoal" status="run" time="1.299" classname="CHOMPMoveitTest" />
    <testcase name="noStartState" status="run" time="1.312" classname="CHOMPMoveitTest">
      <failure message="/root/ws_moveit/src/moveit/moveit_planners/chomp/chomp_interface/test/chomp_moveit_test.cpp:96&#x0A;Value of: moveit::planning_interface::MoveItErrorCode::INVALID_ROBOT_STATE&#x0A;  Actual: -17&#x0A;Expected: error_code.val&#x0A;Which is: 1" type=""><![CDATA[/root/ws_moveit/src/moveit/moveit_planners/chomp/chomp_interface/test/chomp_moveit_test.cpp:96
Value of: moveit::planning_interface::MoveItErrorCode::INVALID_ROBOT_STATE
  Actual: -17
Expected: error_code.val
Which is: 1]]></failure>
    </testcase>
    <testcase name="collisionAtEndOfPath" status="run" time="2.596" classname="CHOMPMoveitTest" />
  </testsuite>
</testsuites>
```

I'm not a good user of `rostest` but I believe that the rationale behind test is contradicting with the behavior implemented as pointed out by @v4hn in one of the recent comments.